test/sde/sde_dynamical.jl

```diff
@@ -3,45 +3,130 @@ Random.seed!(1)
-    display(sim1.𝒪est)
-    @test abs(sim1.𝒪est[:weak_final]-1) < 0.3
+    # # Can't use NoiseGrid as noise is not generated with the correct size in convergence.jl. We require noise with shape of v.
+    # sim1  = analyticless_test_convergence(dts,prob1,BAOAB(gamma=γ),(1/2)^10;trajectories=Int(1e2),use_noise_grid=false)
+    # @test abs(sim1.𝒪est[:weak_final]-1.5) < 0.3
+
+
+    sol1 = solve(prob1,ABOBA(gamma=[γ,γ]);dt=1/10,save_noise=true)
+    prob2 = DynamicalSDEProblem(f1_harmonic_iip,f2_harmonic_iip,g_iip,v0,u0,(0.0,5.0); noise=NoiseWrapper(sol1.W))
+    sol2 = solve(prob2,ABOBA(gamma=[γ,γ]);dt=1/10)
+
+    @test sol1[:] ≈ sol2[:]
+
+    # # Can't use NoiseGrid as noise is not generated with the correct size in convergence.jl. We require noise with shape of v.
+    # sim1  = analyticless_test_convergence(dts,prob1,ABOBA(gamma=γ),(1/2)^10;trajectories=Int(1e4),use_noise_grid=false)
+    # @test abs(sim1.𝒪est[:weak_final]-2) < 0.3
+
+
+    sol1 = solve(prob1,OBABO(gamma=[γ,γ]);dt=1/10,save_noise=true)
+    prob2 = DynamicalSDEProblem(f1_harmonic_iip,f2_harmonic_iip,g_iip,v0,u0,(0.0,5.0); noise=NoiseWrapper(sol1.W))
+    sol2 = solve(prob2,OBABO(gamma=[γ,γ]);dt=1/10)
+
+    @test sol1[:] ≈ sol2[:]
+
+    # # Can't use NoiseGrid as noise is not generated with the correct size in convergence.jl. We require noise with shape of v.
+    # sim1  = analyticless_test_convergence(dts,prob1,OBABO(gamma=γ),(1/2)^10;trajectories=Int(1e4),use_noise_grid=false)
+    # @test abs(sim1.𝒪est[:weak_final]-1.5) < 0.3
 end
 
-@testset "Vector u" begin
+
+
+@testset "Matricial gamma" begin
 
     u0 = zeros(2)
     v0 = ones(2)
 
+    gamma_mat = [γ -0.1*γ;0.5*γ γ]
+
     f1_harmonic_iip(dv,v,u,p,t) = dv .= f1_harmonic(v,u,p,t)
     f2_harmonic_iip(du,v,u,p,t) = du .= f2_harmonic(v,u,p,t)
     g_iip(du,u,p,t) = du .= g(u,p,t)
 
     ff_harmonic = DynamicalSDEFunction(f1_harmonic,f2_harmonic,g)
     prob1 = DynamicalSDEProblem(ff_harmonic,v0,u0,(0.0,5.0))
-    sol1 = solve(prob1,BAOAB(gamma=γ);dt=1/10,save_noise=true)
+    sol1 = solve(prob1,BAOAB(gamma=gamma_mat);dt=1/10,save_noise=true)
 
     prob2 = DynamicalSDEProblem(f1_harmonic_iip,f2_harmonic_iip,g_iip,v0,u0,(0.0,5.0); noise=NoiseWrapper(sol1.W))
-    sol2 = solve(prob2,BAOAB(gamma=γ);dt=1/10)
+    sol2 = solve(prob2,BAOAB(gamma=gamma_mat);dt=1/10)
 
     @test sol1[:] ≈ sol2[:]
 
-    dts = (1/2) .^ (8:-1:4)
 
-    # Can't use NoiseGrid as noise is not generated with the correct size in convergence.jl. We require noise with shape of v.
-    sim1  = analyticless_test_convergence(dts,prob1,BAOAB(gamma=γ),(1/2)^10;trajectories=Int(1e2),use_noise_grid=false)
-    @test abs(sim1.𝒪est[:weak_final]-1) < 0.3
+    sol1 = solve(prob1,ABOBA(gamma=gamma_mat);dt=1/10,save_noise=true)
+    prob2 = DynamicalSDEProblem(f1_harmonic_iip,f2_harmonic_iip,g_iip,v0,u0,(0.0,5.0); noise=NoiseWrapper(sol1.W))
+    sol2 = solve(prob2,ABOBA(gamma=gamma_mat);dt=1/10)
+
+    @test sol1[:] ≈ sol2[:]
```

Comment:
oh I didn't see those tests before -- they are also iip vs oop, right? Please remove then the additional testset I pushed below. (I thought not all of them are tested because it showed that many lines were not covered by the tests in the CI coverage report.)